src/familiar.ts

```diff
@@ -55,177 +53,237 @@ export function meatFamiliar(): Familiar {
   return _meatFamiliar;
 }
 
-function myFamiliarWeight(familiar: Familiar | null = null) {
-  if (familiar === null) familiar = myFamiliar();
-  return familiarWeight(familiar) + weightAdjustment();
+type GeneralFamiliar = {
+  familiar: Familiar;
+  expectedValue: number;
+};
+
+type StandardDropFamiliar = {
+  familiar: Familiar;
+  expected: number[];
+  drop: Item;
+  pref: propertyTypes.NumericProperty;
+  additionalValue?: number;
+};
+
+function valueStandardDropFamiliar({
+  familiar,
+  expected,
+  drop,
+  pref,
+  additionalValue,
+}: StandardDropFamiliar): GeneralFamiliar | null {
+  const expectedTurns = expected[get(pref)];
+  if (!have(familiar) || !expectedTurns) return null;
+  const expectedValue = garboValue(drop) / expectedTurns + (additionalValue ?? 0);
```

Comment:
`expectedValue` here seems to be the expected drop value of a spleener, but a `GeneralFamiliar` only has the field `expectedValue` and not `additionalValue`.

This seems weird, since a general leprechaun should have non-zero expected value (at the very least, it should have a different value from a non-lep).

I prefer the idea of having the following fields instead:
`baseValue`: or synonymously, `lepValue`
`dropValue`: `garboValue(drop) / expectedTurns`
`additionalValue`: For meatify, drones, spit, red present and whatnots (even item% if that ever becomes relevant)
`expectedValue`: `baseValue + dropValue + additionalValue`

Reply:
I think I should cut this down to 
```
familiar: Familiar,
leprechaunMultiplier: number,
additionalValue: number,
```
where additionalValue combines the EV of drops of all sorts. I don't particularly see the value in caching the outfit value as part of the general familiar (I would cache that elsewhere, keyed by leprechaunMultiplier), and `leprechaunMultiplier` should capture all of that lep information you're looking for

Reply:
that being said, the leprechaunMultiplier function lets us derive that from the Familiar itself. But I suppose it's still worth saving as a property of the familiar